debuginfo/src/symbols.rs

```diff
@@ -1,126 +1,202 @@
 use std::collections::{BTreeMap, HashSet};
-use std::iter::Peekable;
-use std::slice::Iter as SliceIter;
+use std::fmt;
+use std::iter::{IntoIterator, Peekable};
+use std::slice;
 
 use goblin::mach;
 
 use symbolic_common::{ErrorKind, Result};
 
 use object::{Object, ObjectTarget};
 
-pub trait SymbolTable {
-    fn symbols(&self) -> Result<Symbols>;
+/// A single symbol
+#[derive(Clone, Copy)]
+pub struct Symbol<'data> {
+    /// Binary string value of the symbol
```

Comment:
can we make the fields private and add an accessor for it so we can easily expand it later?

Reply:
Done